### PR TITLE
Added Model for User

### PR DESCRIPTION
**Pull request by rem-iri**

### CODE REVIEW

**Review by markbugarin — CHANGES_REQUESTED** [review]
use @Data lombok and remove @Builder

File: src/main/java/com/champstart/recipeapp/user/model/Role.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity(name = "roles")
+public class Role {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String name;
+    @ManyToMany(mappedBy = "roles")
```
Comment (markbugarin) [inline src/main/java/com/champstart/recipeapp/user/model/Role.java:21]: Change @ManyToMany into  @OneToOne

File: src/main/java/com/champstart/recipeapp/user/model/Role.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity(name = "roles")
+public class Role {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String name;
```
Comment (QueenElle) [inline src/main/java/com/champstart/recipeapp/user/model/Role.java:20]: what variable declaration is this?

File: src/main/java/com/champstart/recipeapp/user/model/Role.java [hunk 3]
```diff
@@ -18,6 +18,6 @@ public class Role {
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String name;
-    @ManyToMany(mappedBy = "roles")
+    @OneToOne(mappedBy = "roles")
     private List<UserModel> users = new ArrayList<>();
```
Comment (markbugarin) [inline src/main/java/com/champstart/recipeapp/user/model/Role.java:22]: dont use List when using @OneToOne

**Review by markbugarin — APPROVED** [review]

**Review by QueenElle — APPROVED** [review]

**Review by leonardnarvaez — APPROVED** [review]
Approved

**Pull request merged**